CONTRIBUTING.md

```diff
@@ -0,0 +1,267 @@
+To enforce this, all pull requests that must pass [ESLint](http://eslint.org/)
+before they can merge.
+
+#### Markdown
+
+In addition to enforcing a JavaScript style guide, we also require that markdown
+files pass [remarklint](https://github.com/wooorm/remark-lint) with the recommended
+preset. This helps keep our markdown tidy, consistent, and compatible with a range of
+markdown parsers used for generating documentation.
+
+### Node.js Version Requirements
+
+Lux is built against Node `>= v6`. Since this is the latest LTS release and the
+version we run in our CI environments, we recommend you use it when working on
+the Lux codebase.
+
+If you use [nvm](https://github.com/creationix/nvm) to manage Node.js versions
+and zsh (like [Oh-My-ZSH](https://github.com/robbyrussell/oh-my-zsh)), you can
+have nvm switch to the correct Node.js version automatically when you cd into
+this repository. To do so, add the following to your `~/.zshrc` file:
+
+```bash
+# place this after nvm initialization!
+autoload -U add-zsh-hook
+load-nvmrc() {
+  local node_version="$(nvm version)"
+  local nvmrc_path="$(nvm_find_nvmrc)"
+
+  if [ -n "$nvmrc_path" ]; then
+    local nvmrc_node_version=$(nvm version "$(cat "${nvmrc_path}")")
+
+    if [ "$nvmrc_node_version" != "N/A" ] && [ "$nvmrc_node_version" != "$node_version" ]; then
+      nvm install
+    fi
+  elif [ "$node_version" != "$(nvm version default)" ]; then
+    echo "Reverting to nvm default version"
+    nvm use default
+  fi
+}
+add-zsh-hook chpwd load-nvmrc
+load-nvmrc
+```
+
+## Writing Documentation
+
+Improvements to documentation are a great way to start contributing to Lux. The
+sources for the official documentation is generated from source code comments
+and markdown files that live in this repository.
+
+The [Guide](https://lux.postlight.com/docs/latest/guide) section of the official
+documentation is generated from markdown files found in files within the[`./guide`](./guide)
+directory of this repository.
+
+The [API reference](https://lux.postlight.com/docs/latest/api) section of the official
+documentation is generated from source code comments found in files within the [`./src`](./src)
+directory of this repository.
+
+## Submitting a Pull Request
+
+Want to make a change to Lux? Submit a pull request! We use the "fork and pull"
+model [described here](https://help.github.com/articles/creating-a-pull-request-from-a-fork).
+
+**Before submitting a pull request**, please make sure:
+
+*   [X] You have added tests for modifications you made to the codebase.
+*   [X] You have updated any documentation in the source code comments for APIs
+that you may have changed.
+*   [X] You have no linter errors to correct after running `npm run lint`.
+*   [X] You have no type errors to correct after running `npm run flow`.
+*   [X] You have run the test suite via `npm test` and it passed.
+
```

Comment:
This is a good point. I'm not sure that we have to explicitly mention it since GitHub introduced the squash and merge button. I'm open to discussing this more.